stock_picking_accounting_date/models/stock_move.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023-2024 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    accounting_date = fields.Date(
+        compute="_compute_accounting_date",
+        store=True,
+    )
+
+    @api.depends("date", "accounting_date")
+    def _compute_accounting_date(self):
+        # 'force_period_date' context is assigned when user sets accounting date in
+        # inventory adjustment
+        force_period_date = self._context.get("force_period_date")
+        if force_period_date:
+            for rec in self:
+                rec.accounting_date = force_period_date
+        else:
+            for rec in self:
+                if rec.picking_id.accounting_date:
+                    rec.accounting_date = rec.picking_id.accounting_date
+                    continue
+                rec.accounting_date = fields.Datetime.context_timestamp(self, rec.date)
```

Comment:
Let's remove the condition based on the context here, as @rousseldenis suggests.
```suggestion
        for rec in self:
            if rec.picking_id.accounting_date:
                rec.accounting_date = rec.picking_id.accounting_date
                continue
            rec.accounting_date = fields.Datetime.context_timestamp(self, rec.date)
```